src/Examine.Core/Search/IOrdering.cs

```diff
@@ -37,5 +38,12 @@ public interface IOrdering : IQueryExecutor
         /// </summary>
         /// <returns></returns>
         IOrdering SelectAllFields();
+
+        /// <summary>
+        /// Allows for selecting facets to return in your query
+        /// </summary>
+        /// <param name="facets"></param>
+        /// <returns></returns>
+        IQueryExecutor WithFacets(Action<IFaceting> facets);
```

Comment:
This is a binary breaking change too. We could avoid this with extension methods and another interface:

* New interface that exposes: `IQueryExecutor WithFacets(Action<IFaceting> facets);` - could be `IFaceting` and the current `IFaceting` could be changed to `IFacetOperations`
* An extension method on IOrdering with this same signature: `IQueryExecutor WithFacets(Action<IFaceting> facets);`
* Within that extension method, check if the instance of IOrdering is IFaceting, else throw NotSupportedException

We could also have another ext method like `SupportsFacets` that does this same check in case developers want to check if the Examine implementation supports it.

Reply:
I've implemented this now.